packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -615,11 +770,36 @@ class _MonthPickerState extends State<_MonthPicker> {
   }
 }
 
+/// InheritedWidget indicating what the current focused date is for its children.
+///
+/// This is used by the [_MonthPicker] to let its children [_DayPicker]s what
+/// the currently focused date (if any) should be.
+class _FocusedDate extends InheritedWidget {
+  const _FocusedDate({
+    Key key,
+    Widget child,
+    this.date
+  }) : super(key: key, child: child);
+
+  final DateTime date;
+
+  @override
+  bool updateShouldNotify(InheritedWidget oldWidget) {
```

Comment:
Could `oldWidget` not be strongly typed to `_FocusedDate` like I've seen in other InheritedWidgets? Then you wouldn't need to check the type

Reply:
Yup. Good catch. Thx.